pkg/streamline/synchronizer.go

```diff
@@ -181,6 +190,39 @@ func (in *synchronizer) handleEvent(ev watch.Event) {
 	}
 }
 
+func (in *synchronizer) maybeSyncServiceComponents(resource unstructured.Unstructured, isDeleted bool) {
```

Comment:
this will definitely need to be backgrounded somehow.  Ideally you just have a workqueue you throw the updates on and a goroutine or a set of goroutines polling it to flush updates.